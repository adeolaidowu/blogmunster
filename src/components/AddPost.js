import React from "react";

const AddPost = () => <div>This is my create post page</div>;

export default AddPost;
