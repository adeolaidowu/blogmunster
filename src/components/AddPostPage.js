import React from "react";
import { connect } from "react-redux";
import PostForm from "./PostForm";
import { startAddPost } from "../actions/posts";

export const AddPostPage = (props) => {
  const onSubmit = (post) => {
    props.startAddPost(post);
    props.history.push("/");
  };
  return (
    <div>
      <h1>Create Blog Post</h1>
      <PostForm onSubmit={onSubmit} />
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startAddPost: (post) => dispatch(startAddPost(post)),
});

export default connect(undefined, mapDispatchToProps)(AddPostPage);
