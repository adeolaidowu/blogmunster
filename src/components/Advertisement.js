import React from "react";

const Advertisement = () => (
  <div className="text-center">
    <h2>Advertise with us</h2>
    <img src="https://place-hold.it/200x200" alt="placeholder-img" />
    <p>Call 080-XXX-XXX-36</p>
  </div>
);

export default Advertisement;
