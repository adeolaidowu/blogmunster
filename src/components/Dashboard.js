import React from "react";
import { Link } from "react-router-dom";
import PostList from "./PostList";
import PostFilters from "./PostFilters";
import Sidebar from "./Sidebar";

export const Dashboard = () => (
  <div className="dashboard">
    <div className="dashboard__header">
      <h1>My Dashboard</h1>
      <Link className="button btn-primary" to="/create">
        Create New Post
      </Link>
    </div>
    <PostFilters />
    <div className="dashboard__content">
      <PostList />
    </div>
  </div>
);

export default Dashboard;
