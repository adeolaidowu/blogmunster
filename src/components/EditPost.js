import React from "react";
import { connect } from "react-redux";
import PostForm from "./PostForm";
import { editPost, deletePost } from "../actions/posts";

const EditPost = (props) => {
  const onSubmit = (post) => {
    props.dispatch(editPost(props.post.id, post));
    props.history.push("/");
  };
  const onDelete = () => {
    props.dispatch(deletePost({ id: props.post.id }));
    props.history.push("/");
  };
  return (
    <div>
      <PostForm post={props.post} onSubmit={onSubmit} />
      <button onClick={onDelete}>Delete</button>
    </div>
  );
};
const mapStateToProps = (state, props) => ({
  post: state.posts.find((post) => props.match.params.id === post.id),
});

export default connect(mapStateToProps)(EditPost);
