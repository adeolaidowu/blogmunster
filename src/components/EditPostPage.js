import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import PostForm from "./PostForm";
import { startEditPost, startDeletePost } from "../actions/posts";

export const EditPostPage = (props) => {
  const onSubmit = (post) => {
    props.startEditPost(props.post.id, post);
    props.history.push("/dashboard");
  };
  const onDelete = () => {
    props.startDeletePost({ id: props.post.id });
    props.history.push("/dashboard");
  };
  return (
    <div className="page-container">
      <div className="page-header">
        <Link to="/dashboard" className="button">
          Dashboard
        </Link>
        <button className="button btn-danger" onClick={onDelete}>
          Delete Post
        </button>
      </div>
      <h1>Edit Blog Post</h1>
      <PostForm post={props.post} onSubmit={onSubmit} />
    </div>
  );
};
const mapStateToProps = (state, props) => ({
  post: state.posts.find((post) => props.match.params.id === post.id),
});

const mapDispatchToProps = (dispatch) => ({
  startEditPost: (id, updates) => dispatch(startEditPost(id, updates)),
  startDeletePost: (data) => dispatch(startDeletePost(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPostPage);
