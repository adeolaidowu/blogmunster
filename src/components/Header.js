import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogin, startLogout } from "../actions/auth";

export const Header = ({ startLogin, startLogout, uid }) => {
  return (
    <header className="header">
      <div className="content-container">
        <div className="header__content">
          <Link to="/" className="header__title">
            <h1>BLOGMUNSTER</h1>
          </Link>
          {uid ? (
            <button className="button" onClick={startLogout}>
              Logout
            </button>
          ) : (
            <button className="button" onClick={startLogin}>
              Login with Google
            </button>
          )}
        </div>
      </div>
    </header>
  );
};

const mapStateToProps = (state) => ({
  uid: state.auth.uid,
});

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
