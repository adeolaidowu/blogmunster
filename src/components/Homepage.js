import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
import PostList from "./PostList";
import PostFilters from "./PostFilters";

export const Homepage = (props) => {
  return (
    <div>
      <h1>Welcome to BlogMunster</h1>
      <button onClick={props.startLogin}>Login with Google</button>
      {<PostFilters />}
      {<PostList />}
    </div>
  );
};

// const mapStateToProps = (state) => ({
//   presentState: state,
// });
const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(Homepage);
