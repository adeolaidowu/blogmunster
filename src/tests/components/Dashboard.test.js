import React from "react";
import { Dashboard } from "../../components/Dashboard";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

test("should correctly render component", () => {
  const wrapper = shallow(<Dashboard startLogout={() => {}} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should call startLogout when clicked", () => {
  const startLogoutSpy = jest.fn();
  const wrapper = shallow(<Dashboard startLogout={startLogoutSpy} />);
  wrapper.find("button").simulate("click");
  expect(startLogoutSpy).toHaveBeenCalled();
});
