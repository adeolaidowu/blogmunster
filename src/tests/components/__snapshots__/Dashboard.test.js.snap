// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render component 1`] = `
<div>
  <h1>
    My Dashboard
  </h1>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
  <NavLink
    to="/create"
  >
    Create New Post
  </NavLink>
  <Connect(PostFilters) />
  <Connect(PostList) />
</div>
`;
